solutions/cf-1555B.mdx

```diff
@@ -0,0 +1,341 @@
+If the first table were to move in a diagonal motion, the table could achieve the same result just moving in the x or y direction. 
+(The hypotenuse - the diagonal motion - will always be longer then the other sides - the x or y motion)
+
+Let's call: 
+* $x_{1}$ and $x_{2}$ the leftmost and rightmost positions of the inital rectangle respectively
+* $y_{1}$ and $y_{2}$ the bottommost and topmost positions of the inital rectangle respectively
+* $w_{1}$ and $h_{1}$ the width and height of the inital rectangle (which can be calculated by calulating the difference in x and y)
+* $w_{2}$ and $h_{2}$ the width and height of the second rectangle respectively
+* $W$ and $H$ the total width and height of the room
+
+More importantly, there are 4 different places the second rectangle could be placed: left, right, top, bottom.
+For each place, we can calculate the correct x or y positions in order to just barely fit the second rectangle.
+
+* To put the second rectangle on the left side, we just need to place the first rectangle's leftmost position or $x_{1}$ to at least $w_{2}$.
+* To put the second rectangle on the right side, we just need to place the first rectangle's rightmost position or $x_{2}$ to at least $W$ - $w_{2}$
+* To put the second rectangle on the top side, we just need to place the first rectangle's highest position or $y_{2}$ to at least $W$ - $w_{2}$
+* To put the second rectangle on the bottom side, we just need to place the first rectangle's lowest position or $y_{1}$ to at least $w_{2}$
+
+Note: when placing the second rectangle on the left and right sides, make sure $w_{1} + w_{2} <= W$, otherwise, the second rectangle won't fit.
+Also, when placing the second rectangle on the top and bottom sides, make sure $h_{1} + h_{2} <= H$, otherwise, the second rectangle won't fit either of them.
+
+Calculate these placements, and chose the placement that requires the least distance.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(1)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+void solve()
+{
+	int totalWidth, totalHeight; 
+	cin >> totalWidth >> totalHeight;
+	
+	int x1, y1, x2, y2; 
+	cin >> x1 >> y1 >> x2 >> y2;
+	
+	int w1 = x2 - x1;
+	int h1 = y2 - y1;
+	
+	
+	int w2, h2; 
+	cin >> w2 >> h2;
+	
+
+	/*These are the optimal placements to just
+	barely fit the second rectangle */
+	
+	//minimum X value needed to fit on left
+	int leftPlace = w2;
+	
+	//maximum Y value needed to fit on the right
+	int rightPlace = totalWidth - w2;
+
+	//the maximum Y value needed to fit on top
+	int topPlace = totalHeight - h2;
+	
+	//minimum Y value needed to fit on the bottom
+	int bottomPlace = h2;
+
+	/*
+		Going through all possibilities:
+		Top, Bottom, Left, Right
+		Calculate the distance needed to move
+		to the corresponding places.
+		
+		If that distance is negative, that means we don't have
+		to move the initial rectangle
+	*/

```

Comment:
see above

Reply:
also if it's supposed to be called `top left, top right, bottom ...` update here as well